ntoskrnl/mm/ARM3/virtual.c

```diff
@@ -2034,12 +2099,18 @@ MiIsEntireRangeCommitted(IN ULONG_PTR StartingAddress,
             }
             else
             {
+                /* Is the entire VAD committed? If not, fail */
+                if (!Vad->u.VadFlags.MemCommit) return FALSE;
```

Comment:
Question: Can page tables be decommitted, if the entire range of PTEs is decommitted?

Reply:
Yes. Page tables must be accounted for when calculating commit charges. So each time we commit a PTE, we must check if the PDE/PPE/PXE supporting it is, and if not, act accordingly.
Same thing when decommiting, but there it gets a bit more complicated, one PDE can be underneath more than one committed range.

Reply:
Related to this: https://github.com/tkreuzer/reactos/commit/d002afa2cae8439a200cbdb88bea6aee3117108b#r51788177